### PR TITLE
Effect latencies

### PR DESCRIPTION
**Pull request by Paul-Licameli**
Resolves: #3221 

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: src/effects/RealtimeEffectManager.cpp [hunk 1]
```diff
@@ -129,7 +129,7 @@ size_t RealtimeEffectManager::Process(bool suspended, Track &track,
    // Can be suspended because of the audio stream being paused or because effects
```
Comment (pietro68) [inline src/effects/RealtimeEffectManager.cpp:129]: So the meaning of the return value has changed; but it is difficult for me to understand what it is now. Number of processed samples? a comment in the prototype would help

File: src/AudioIO.cpp [hunk 2]
```diff
@@ -1795,22 +1795,22 @@ void AudioIO::FillPlayBuffers()
...
+   // It is possible that some buffers will have more samples available than
+   // others.  This could happen if we hit this code during the PortAudio
+   // callback.  Also, if in a previous pass, unequal numbers of samples were
+   // discarded from ring buffers for differing latencies.
+
+   // To keep things simple, we write no more data than is vacant in
    // ALL buffers, and advance the global time by that much.
    auto nAvailable = GetCommonlyFreePlayback();
 
    // Don't fill the buffers at all unless we can do the
```
Comment (pietro68) [inline src/AudioIO.cpp:1807]: maybe "the" should be deleted?

File: src/effects/RealtimeEffectManager.cpp (hunk 1 above)
Comment (Paul-Licameli) [reply src/effects/RealtimeEffectManager.cpp:129]: Exactly.  The comments in header files are updated.

The sum of latencies of the effects must be discarded but the state object is responsible to report the count, only, not do the discarding.

**Review by pietro68 — APPROVED** [review]
I can not honestly say that I fully understood the changes, but I get a sense of them - I do not know this part of the code well enough. Anyway, I verified that the fix works.

**Pull request merged**